src/components/Breadcrumb.vue

```diff
@@ -23,44 +29,71 @@ export default {
   },
   asyncComputed: {
     breadcrumb: {
-      default: { root: {}, path: [] },
+      default: [],
```

Comment:
Rather than the very long conditional block in the template,  I think much of this logic could be simplified by making the breadcrumb schema:

`{ filespace: enum('users', 'collections', null), path: [] }`

The default would be:

`{ filespace: null, path: [] }`

* "root" is static and determined by the value of noRoot
* "fileSpace" will always be one of `'users'`, `'collections'`, `null`
* "path" can have the exact same logic as before.

Reply:
Honestly, the conditional in the template is long but not complex. so I don't think introducing new concepts is simplifying.

Reply:
Fewer conditionals in the template improve testability.  There's enough branching going on in the template that if we moved it into the computed prop itself, it would be a lot easier to get coverage on.